tests/test_flows.py

```diff
@@ -201,6 +201,70 @@ def check_cmdu_has_tlvs_exact(
             result = False
         return result
 
+    def check_topology_notification(self, eth_src: str, neighbors: list,
+                                    sta: env.Station, event: env.StationEvent, bssid: str) -> bool:
+        """Verify topology notification reliable multicast - given a source mac and
+           a list of neighbors macs, check that exactly one relayed multicast CMDU
+           was sent to the IEEE1905.1 multicast MAC address, and a single unicast
+           CMDU with the relayed bit unset to each of the given neighbors destination MACs.
+           Verify correctness of the association event TLV inside the topology notification.
+           Mark failure if any of the above conditions isn't met.
+
+        Parameters
+        ----------
+
+        eth_src: str
+            source AL MAC (origin of the topology notification)
+
+        neighbors: list
+            destination AL MACs (destinations of the topology notification)
+
+        sta: environment.Station
+            station mac
+
+        event: environment.StationEvent
+            station event - CONNECTED / DISCONNECTED
+
+        bssid: str
+            bssid Multi-AP Agent BSSID
+
+        Returns:
+        bool
+            True for valid topology notification, False otherwise
+        """
+        multicasts = self.check_cmdu_type("topology notification", 0x1, eth_src)
+
+        # check correctness of topology notifications
+        for multicast in multicasts:
+            # relay indication should be set
+            if not multicast.ieee1905_relay_indicator:
+                self.fail("Multicast topology notification should be relayed")
+                return False
+
+            mid = multicast.ieee1905_mid
+            for eth_dst in neighbors:
+                unicasts = self.check_cmdu_type("topology notification", 0x1, eth_src, eth_dst)
+                matches = [unicast for unicast in unicasts if unicast.ieee1905_mid == mid]
+                if len(matches) != 1:
+                    self.fail("Recieved {} topology notifications to {} with mid {} (should be one)"
+                              .format(len(matches), eth_dst, mid))
+                    return False
+                unicast = matches[0]
+                if unicast.ieee1905_relay_indicator:
```

Comment:
We could actually add that check to check_cmdu_type: if it's not multicast, it should never have the relay indicator.

Reply:
Is this what you mean?
```
if eth_dst is None:
            eth_dst = "01:80:c2:00:00:13"
        return self.check_cmdu(msg, lambda packet:
                               packet.eth_src == eth_src and
                               packet.eth_dst == eth_dst and
                               packet.ieee1905_message_type == msg_type and
                               (mid is None or packet.ieee1905_mid == mid) and
                               (eth_dst != "01:80:c2:00:00:13" or
                                packet.ieee1905_relay_indicator is False))
```

Reply:
Cause it doesn't work :/

Reply:
I was thinking more in check_cmdu itself:
```
for packet in result:
    if packet.ieee1905:
        if packet.eth_dst != "01:80:c2:00:00:13" and packet.ieee1905_relay_indicator:
            self.fail("Non-multicast CMDU with relay indicator set: {}".format(packet))
```

Could even be done on the full capture instead of the result, but then you'd get the same error several times most likely.